packages/grafana-ui/src/components/GraphNG/utils.ts

```diff
@@ -219,8 +223,28 @@ export function preparePlotConfigBuilder(
       fieldName: getFieldDisplayName(field, frame),
       hideInLegend: customConfig.hideFrom?.legend,
     });
+
+    if (customConfig?.stackingMode !== StackingMode.None && customConfig?.stackingGroup) {
+      if (!stackingGroups[customConfig.stackingGroup]) {
+        stackingGroups[customConfig.stackingGroup] = [seriesIndex];
+      } else {
+        stackingGroups[customConfig.stackingGroup].push(seriesIndex);
+      }
+    }
   }
 
+  if (Object.keys(stackingGroups).length !== 0) {
+    builder.setStacking(true);
+    const groups = Object.keys(stackingGroups);
+    for (let i = 0; i < groups.length; i++) {
+      const group = stackingGroups[groups[i]];
+      for (let j = group.length - 1; j > 0; j--) {
+        builder.addBand({
+          series: [group[j], group[j - 1]],
+        });
+      }
+    }
+  }
```

Comment:
Maybe we should extend or add tests for this? If I was ever to change things in this code I would feel a lot safer if there were tests.